tests/model_serving/model_runtime/vllm/toolcalling/test_granite_3_2_8b_instruct_preview.py

```diff
@@ -0,0 +1,201 @@
+import pytest
+from simple_logger.logger import get_logger
+from typing import List, Any, Generator
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.pod import Pod
+from utilities.constants import KServeDeploymentType
+from tests.model_serving.model_runtime.vllm.utils import (
+    run_raw_inference,
+    validate_inference_output,
+)
+from tests.model_serving.model_runtime.vllm.constant import (
+    OPENAI_ENDPOINT_NAME,
+    TGIS_ENDPOINT_NAME,
+    LIGHTSPEED_TOOL_QUERY,
+    LIGHTSPEED_TOOL,
+    WEATHER_TOOL,
+    WEATHER_TOOL_QUERY,
+    MATH_CHAT_QUERY,
+)
+
+LOGGER = get_logger(name=__name__)
+
+
+SERVING_ARGUMENT: List[str] = [
+    "--model=/mnt/models",
+    "--uvicorn-log-level=debug",
+    "--dtype=float16",
+    "--chat-template=/app/data/template/tool_chat_template_granite.jinja",
+    "--enable-auto-tool-choice",
+    "--tool-call-parser=granite",
+]
+
+MODEL_PATH: str = "ibm-granite/granite-3.2-8b-instruct-preview"
+
+
+pytestmark = pytest.mark.usefixtures("skip_if_no_supported_accelerator_type", "valid_aws_config")
+
+
+@pytest.mark.parametrize(
+    "model_namespace, s3_models_storage_uri, serving_runtime, vllm_inference_service",
+    [
+        pytest.param(
+            {"name": "granite-32-8b"},
+            {"model-dir": MODEL_PATH},
+            {"deployment_type": KServeDeploymentType.RAW_DEPLOYMENT},
+            {
+                "deployment_mode": KServeDeploymentType.RAW_DEPLOYMENT,
+                "runtime_argument": SERVING_ARGUMENT,
+                "gpu_count": 1,
+                "name": "granite-32-8b",
+                "min-replicas": 1,
+            },
+        ),
+    ],
+    indirect=True,
+)
+class TestGranite32ToolModel:
+    def test_granite_model_inference(
+        self,
+        vllm_inference_service: Generator[InferenceService, Any, Any],
+        get_pod_name_resource: Pod,
```

Comment:
`get_pod_name_resource` was not added in the pr but should be renamed, a fixture name should be a noun (in this case something like `vllm_pod` ) and not a verb

Reply:
I've opened an issue for this in the repo. Once this PR is merged, I will make the requested changes for all the vLLM tests in single PR